File: src/main/java/maple/expectation/controller/dto/auth/TokenResponse.java
package maple.expectation.controller.dto.auth;

/**
 * Token Refresh 응답 DTO (Issue #279)
 *
 * @param accessToken       새 Access Token (JWT)
 * @param accessExpiresIn   Access Token 만료 시간 (초)
 * @param refreshToken      새 Refresh Token ID
 * @param refreshExpiresIn  Refresh Token 만료 시간 (초)
 */
public record TokenResponse(
    String accessToken,
    long accessExpiresIn,
    String refreshToken,
    long refreshExpiresIn
) {
    public static TokenResponse of(String accessToken, long accessExpiresIn,
                                   String refreshToken, long refreshExpiresIn) {
        return new TokenResponse(accessToken, accessExpiresIn, refreshToken, refreshExpiresIn);
    }
}

File: module-app/src/main/kotlin/maple/expectation/controller/dto/auth/TokenResponse.kt
package maple.expectation.controller.dto.auth

/**
 * Token Refresh 응답 DTO (Issue #279)
 *
 * @property accessToken 새 Access Token (JWT)
 * @property accessExpiresIn Access Token 만료 시간 (초)
 * @property refreshToken 새 Refresh Token ID
 * @property refreshExpiresIn Refresh Token 만료 시간 (초)
 */
data class TokenResponse(
    val accessToken: String,
    val accessExpiresIn: Long,
    val refreshToken: String,
    val refreshExpiresIn: Long
) {
    // Explicit Record-style accessors for production code compatibility
    fun accessToken(): String = accessToken
    fun accessExpiresIn(): Long = accessExpiresIn
    fun refreshToken(): String = refreshToken
    fun refreshExpiresIn(): Long = refreshExpiresIn

    companion object {
        @JvmStatic
        fun of(
            accessToken: String,
            accessExpiresIn: Long,
            refreshToken: String,
            refreshExpiresIn: Long
        ): TokenResponse = TokenResponse(accessToken, accessExpiresIn, refreshToken, refreshExpiresIn)
    }
}
